File: backend/api/src/main/java/fr/cpe/sdmis/domain/model/Evenement.java
package fr.cpe.sdmis.domain.model;

import java.util.UUID;

public record Evenement(
        UUID id,
        String description,
        double latitude,
        double longitude,
        UUID idTypeEvenement,
        UUID idStatut,
        UUID idSeverite,
        String nomTypeEvenement,
        String nomStatut,
        String nomSeverite,
        String valeurEchelle,
        Integer nbVehiculesNecessaire
) { }

File: frontend/src/models/evenement.ts
export interface EvenementApi {
  id: string
  description: string
  latitude: number
  longitude: number
  idTypeEvenement: string
  idStatut: string
  idSeverite: string
  nomTypeEvenement: string
  nomStatut: string
  nomSeverite: string
  valeurEchelle: string
  nbVehiculesNecessaire: number | null
}

export interface EvenementCreatePayload {
  description: string
  latitude: number
  longitude: number
  idTypeEvenement: string
  idStatut: string
  idSeverite: string
}

export interface SeveriteReference {
  id: string
  nomSeverite: string
  valeurEchelle: string
  nbVehiculesNecessaire: number | null
}

export interface TypeEvenementReference {
  id: string
  nomTypeEvenement: string
}
